pkg-manager/resolve-dependencies/src/getNonDevWantedDependencies.ts

```diff
@@ -7,6 +7,7 @@ export interface WantedDependency {
   dev: boolean
   optional: boolean
   injected?: boolean
+  source?: 'cli-param'
```

Comment:
The resolver shouldn't care about the source. It should just care if the dependency should be saved to catalogs.

```suggestion
  saveCatalog?: boolean
```

Reply:
https://github.com/pnpm/pnpm/pull/9484/commits/5a4959d130af4e5f4ed8ebf6cbfae5c75a565ea2